### PR TITLE
Check if ports are accessible via external service

### PR DESCRIPTION
**Pull request by soffokl**

### CODE REVIEW

File: p2p/nat/upnp.go [hunk 1]
```diff
@@ -79,6 +79,11 @@ func (up *upnpPort) PreparePorts() (ports []int, release func(), start StartPort
 		ports = append(ports, p.Num())
 	}
 
+	if err := checkAllPorts(ports); err != nil {
+		log.Debug().Err(err).Msgf("Failed to check manual ports %d globally", ports)
```
Comment (Snawoot) [inline p2p/nat/upnp.go:83]:
```suggestion
		log.Debug().Err(err).Msgf("Failed to check UPnP ports %d globally", ports)
```

**Review by Snawoot — APPROVED** [review]

File: config/flags_network.go [hunk 2]
```diff
@@ -118,7 +118,13 @@ var (
 	FlagTraversal = cli.StringFlag{
 		Name:  "traversal",
 		Usage: "Comma separated order of NAT traversal methods to be used for providing service",
-		Value: "holepunching,upnp,manual",
+		Value: "manual,upnp,holepunching",
+	}
+	// FlagPortCheckServers list of asymmetric UDP echo servers for checking port availability
+	FlagPortCheckServers = cli.StringFlag{
```
Comment (vkuznecovas) [inline config/flags_network.go:124]: make this hidden, users will have little clue as to what it is and are unlikely to use this, ever. This just adds another flag to an already confusing list :)

**Review by vkuznecovas — APPROVED** [review]

File: p2p/nat/check.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+func checkAllPorts(ports []int) error {
+	var addresses []string
+
+	for _, address := range strings.Split(config.GetString(config.FlagPortCheckServers), ",") {
+		address = strings.TrimSpace(address)
+		if address != "" {
+			addresses = append(addresses, address)
+		}
+	}
+
```
Comment (tomasmik) [inline p2p/nat/check.go:52]: Add a check to see if you even have any addresses to use? Not point to continue if not.

**Review by tomasmik — APPROVED** [review]

**Review by mdomasevicius — APPROVED** [review]

File: p2p/nat/check.go (hunk 3 above)
Comment (soffokl) [reply p2p/nat/check.go:52]: We already have this check in a level deeper.

**Pull request merged**